src/server/json_family.cc

```diff
@@ -872,8 +903,77 @@ OpResult<vector<OptString>> OpMGet(const OpArgs& op_args, const vector<string_vi
   return vec;
 }
 
+// Returns numeric vector that represents the number of fields of JSON value at each path.
+OpResult<vector<OptSizeT>> OpFields(const OpArgs& op_args, string_view key,
+                                    JsonExpression expression) {
+  OpResult<json> result = GetJson(op_args, key);
+  if (!result) {
+    return result.status();
+  }
+
+  vector<OptSizeT> vec;
+  auto cb = [&vec](const string_view& path, const json& val) {
+    if (val.is_array() || val.is_object()) {
+      vec.emplace_back(CountJsonFields(val));
+    } else {
+      vec.emplace_back(1);
+    }
```

Comment:
Doesn't CountJsonFields already return the correct value for non object and non array types?

Reply:
According to your suggestion, I amended the loop body.